### PR TITLE
wallet: AvailableCoins, simplify output script type acquisition

### PR DESCRIPTION
**Pull request by furszy**
There is an unnecessary `ExtractDestination()` call and subsequent result parse into an `CScriptID`.

The `Solver()` call, which we are already doing below anyway, retrieves the script type and, in the P2SH case, the program id.

### CODE REVIEW

**Comment by theStack** [comment]
Concept ACK

**Review by aureleoules — COMMENTED** [review]
ACK 24c82ee0987a39a551f9ab40b95b72e3ed1e224f - LGTM

File: src/wallet/spend.cpp [hunk 1]
```diff
@@ -260,32 +260,24 @@ CoinsResult AvailableCoins(const CWallet& wallet,
...
-            CScript script;
             bool is_from_p2sh{false};
-            if (output.scriptPubKey.IsPayToScriptHash() && solvable) {
-                CTxDestination destination;
-                if (!ExtractDestination(output.scriptPubKey, destination))
-                    continue;
-                const CScriptID& hash = CScriptID(std::get<ScriptHash>(destination));
-                if (!provider->GetCScript(hash, script))
-                    continue;
+            if (type == TxoutType::SCRIPTHASH && solvable) {
```
Comment (rajarshimaitra) [inline src/wallet/spend.cpp:272]: In the comment above it is mentioned that we check this if the output is "spendable". But in the conditional we are checking for "solvable", which I think is enough to serve the purpose. Should the comment be updated in that light?

**Review by rajarshimaitra — COMMENTED** [review]
tACK 24c82ee0987a39a551f9ab40b95b72e3ed1e224f

Just one non-blocking nit comment not related to changes of this PR.

File: src/wallet/spend.cpp (hunk 1 above)
Comment (furszy) [reply src/wallet/spend.cpp:272]: yeah. The comment isn't accurate.
Solvable refers to the wallet knowing how to create the unlocking script that spends the output (which is what we are looking for here), while spendable refers to whether the wallet has the key/s to sign the input or not.

**Comment by furszy** [comment]
Done, added @rajarshimaitra feedback.

**Review by aureleoules — APPROVED** [review]
re-ACK 58b7df3caa21519de61e10f6ee42f0be9ac3cc30

**Review by rajarshimaitra — APPROVED** [review]
ACK 58b7df3caa21519de61e10f6ee42f0be9ac3cc30

**Review by w0xlt — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/25933/commits/58b7df3caa21519de61e10f6ee42f0be9ac3cc30

**Comment by achow101** [comment]
ACK 58b7df3caa21519de61e10f6ee42f0be9ac3cc30

**Pull request merged**